sources/packages/web/src/components/common/students/StudentRestrictions.vue

```diff
@@ -0,0 +1,240 @@
+<template>
+  <body-header-container>
+    <template #header>
+      <body-header title="All restrictions">
+        <template #actions v-if="canAddRestrictions">
+          <check-permission-role :role="Role.StudentAddRestriction">
+            <template #="{ notAllowed }">
+              <v-btn
+                @click="addStudentRestriction"
+                class="float-right"
+                color="primary"
+                data-cy="addRestrictionButton"
+                prepend-icon="fa:fa fa-plus-circle"
+                :disabled="notAllowed"
+                >Add restriction</v-btn
+              >
+            </template>
+          </check-permission-role>
+        </template>
+      </body-header>
+    </template>
+    <content-group>
+      <DataTable
+        :value="studentRestrictions"
+        :paginator="true"
+        :rows="DEFAULT_PAGE_LIMIT"
+        :rowsPerPageOptions="PAGINATION_LIST"
+      >
+        <template #empty>
+          <p class="text-center font-weight-bold">No records found.</p>
+        </template>
+        <Column
+          field="restrictionCategory"
+          header="Category"
+          :sortable="true"
+        ></Column>
+        <Column field="description" header="Reason">
+          <template #body="slotProps">{{
+            `${slotProps.data.restrictionCode} - ${slotProps.data.description}`
+          }}</template></Column
+        >
+        <Column field="createdAt" header="Added"
+          ><template #body="slotProps">{{
+            dateOnlyLongString(slotProps.data.createdAt)
+          }}</template></Column
+        >
+        <Column field="updatedAt" header="Resolved">
+          <template #body="slotProps">{{
+            slotProps.data.isActive
+              ? "-"
```

Comment:
As a suggestion, the `emptyStringFiller` can be adapted for this situation also.

Reply:
Had a conversation with @dheepak-aot and since the passed argument and the returned value to the `emptyStringFiller` function differ from each other in this case, got the suggestion to create a new helper extending `emptyStringFiller`

Reply:
I can see that. Maybe my suggestion was not clear but the comment states to use the `emptyStringFiller ` and have it adapted (emphasis on the adapt part).
I am ok with the new method.